test/acceptance/tests/hcp/hcp_test.go

```diff
@@ -1,124 +1,119 @@
 package hcp
 
 import (
-	"encoding/json"
+	"context"
 	"fmt"
 	"strings"
 	"testing"
 	"time"
 
 	terratestLogger "github.com/gruntwork-io/terratest/modules/logger"
 	"github.com/gruntwork-io/terratest/modules/random"
-	"github.com/gruntwork-io/terratest/modules/shell"
 	"github.com/gruntwork-io/terratest/modules/terraform"
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/terraform-aws-consul-ecs/test/acceptance/framework/config"
 	"github.com/hashicorp/terraform-aws-consul-ecs/test/acceptance/framework/helpers"
 	"github.com/hashicorp/terraform-aws-consul-ecs/test/acceptance/framework/logger"
 	"github.com/stretchr/testify/require"
 )
 
+const setupDir = "../../setup-terraform"
+
+var timeouts = map[string]*retry.Timer{
+	// Timeout and polling interval for ECS mesh tasks to start and register with Consul.
+	"registration": &retry.Timer{Timeout: 6 * time.Minute, Wait: 20 * time.Second},
+	// Timeout and polling interval for ECS mesh tasks to start reporting healthy status.
+	"health": &retry.Timer{Timeout: 3 * time.Minute, Wait: 10 * time.Second},
+	// Timeout and polling interval for making API calls to the Consul server.
+	"consul": &retry.Timer{Timeout: 1 * time.Minute, Wait: 10 * time.Second},
+	// Timeout and polling interval for making service calls (curl) between mesh tasks.
+	"mesh-task": &retry.Timer{Timeout: 5 * time.Minute, Wait: 20 * time.Second},
+}
+
+// HCPTestConfig holds the extended configuration for the admin partition/namespace tests.
+type HCPTestConfig struct {
+	config.TestConfig
+	ECSCluster1ARN          string      `json:"ecs_cluster_1_arn"`
+	ECSCluster2ARN          string      `json:"ecs_cluster_2_arn"`
+	ConsulAddr              string      `json:"consul_public_endpoint_url"`
+	ConsulPrivateAddr       string      `json:"consul_private_endpoint_url"`
+	RetryJoin               interface{} `json:"retry_join"`
+	BootstrapTokenSecretARN string      `json:"bootstrap_token_secret_arn"`
+	GossipKeySecretARN      string      `json:"gossip_key_secret_arn"`
+	ConsulCASecretARN       string      `json:"consul_ca_cert_secret_arn"`
+
+	// Fields that are excluded from the Terraform variables have no JSON tags.
+	ConsulToken string
```

Comment:
> Not that it really matters here, but it will parse a "ConsulToken" field (case insensitively), if present.

Good catch!

In the end we actually need the `token` field to be serialized/deserialized because the tests need it to authenticate the `consulClient`.. this is why the HCP tests failed in the CI pipeline.